Extract clamp mode from nwb file

sgratiy opened the pull request:
At the stage of the feature extraction, the `sweep_record` is missing `clamp_mode` information.
Thus, adding a method for determining the clamp mode. Call it before building a sweep_table.
I had to refactor AibsDataSet and HBGDataSet to factor out individual functions for determining sweep properties.

Resolves #203, #205

njmei commented on tests/test_aibs_data_set.py:
```diff
@@ -0,0 +1,12 @@
+    sweep_info = [{"sweep_number": 0}]
+    dataset = AibsDataSet(sweep_info, nwb_file=NWB_file, api_sweeps=False)
+
+    print list(dataset.sweep_table) == dataset.FILTER_COLUMN_NAMES
```
did you mean `assert` instead of `print`?

sgratiy replied:
good catch

sgratiy commented:
@t-b 
>AIBSDataset: get_stimulus_code and get_stimulus_code_ext both use self.notebook but looking at the constructor this is not always available. Do we need to raise an exception if self.notebook is None?

Yeah, this is a problem. I have significantly modified the PR. Now only the clamp_mode is determined inside the EphysDataSet. The other properties are validated by the ArgSchema anyway. 

I have added the tests as you suggested as well.
Thanks for a careful review.

sgratiy commented:
@nicain 
I have rewritten that function and dropped validation since it is is already done by ArgSchema. I would need to open a separate issue for doing detailed validation though. Thanks for the tip.
